regtests/client/python/cli/command/catalogs.py

```diff
@@ -199,12 +194,33 @@ def execute(self, api: PolarisDefaultApi) -> None:
                     default_base_location=new_default_base_location,
                     additional_properties=new_additional_properties
                 )
-            if (self._has_aws_storage_info() or self._has_azure_storage_info() or self._has_gcs_storage_info() or
-                    self.allowed_locations or self.default_base_location):
+
+            if (self._has_aws_storage_info() or self._has_azure_storage_info() or
+                self._has_gcs_storage_info() or self.allowed_locations):
+                # We must first reconstitute local storage-config related settings from the existing
+                # catalog to properly construct the complete updated storage-config
+                updated_storage_info = catalog.storage_config_info
+
+                # In order to apply mutations client-side, we can't just use the base
+                # _build_storage_config_info helper; instead, each allowed updatable field defined
+                # in option_tree.py should be applied individually against the existing
+                # storage_config_info here.
+                if (self.allowed_locations):
+                    updated_storage_info.allowed_locations.extend(self.allowed_locations)
```

Comment:
Extend, not override?

Reply:
Yeah, I considered override, but because of the way the flag is parsed as a "repetition" of `--allowed_location`, e.g.:

    catalogs create --allowed_location loc1 --allowed_location loc2
    catalogs update --allowed_location loc3

with the semantic that `--allowed_location` is at least additive within a single command, it seemed safer to "extend" for now.

Also, it seems like the most common scenario will be additive, unless tables get unregistered carefully. For removals we might want smarter server-side validation that existing locations aren't violated before we make it too easy to remove items from this config.

Override I agree is more multi-purpose but I'm worried about people accidentally breaking their prod config.

Reply:
I do think the behavior of allowed_locations should probably be better refined in a followup, and we could either change to "override" semantics or add a *different* flag to mean override, and/or add a `--remove-allowed-location` flag.

My original goal here was just to add `--region` update, but I had to fix `--allowed-location` too to make it work.

Reply:
Makes sense to me.